src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripPatternForDates.java

```diff
@@ -111,6 +139,20 @@ public String debugInfo() {
         return numberOfTripSchedules;
     }
 
+    @Override
+    public boolean useCustomizedTripSearch() {
+        return Arrays.stream(tripPatternForDates).anyMatch(TripPatternForDate::hasFrequencies);
```

Comment:
This is relatively performance critical, so we might want to extract it into a field

Reply:
Tested. Was expensive, moved into a field